react-training/src/redux/auth/reducer.js

```diff
@@ -1,37 +1,20 @@
+import { createReducer, onLoading, onSuccess, onFailure } from 'redux-recompose';
+import Immutable from 'seamless-immutable';
+
 import { actionTypes } from '../auth/action';
 
-const initialState = {
-  isAuth: false,
-  user: null,
-  loggingIn: false
+const initialState = Immutable({ auth: null });
+
+const setAuthentication = (state, action) => state.merge({ ...state, [action.target]: action.payload });
+
+const logout = (state, action) => state.merge({ ...state, [action.target]: action.payload });
+
+const reducerDescriptor = {
```

Comment:
Let's use [completeReducer](https://github.com/Wolox/redux-recompose/blob/master/src/completers/completeReducer/docs.md) here to avoid declare actions related to the api integration